src/runtime/methodbinder.cs

```diff
@@ -344,7 +344,7 @@ internal Binding Bind(IntPtr inst, IntPtr args, IntPtr kw)
                             if (clrtype != null)
                             {
                                 bool typematch = false;
-                                if (pi[n].ParameterType != clrtype)
+                                if ((pi[n].ParameterType != clrtype) && !(pi[n].ParameterType.IsAssignableFrom(clrtype)))
```

Comment:
This is a very good catch! But actually the logic is more complicated,
since sometimes multiple methods can match `IsAssignableFrom` criteria for
the overloaded arguments.

So in addition to type equality really need to loop through `IsSubclassOf`
and build a list sorted by proximity to the argument type.

On Thu, May 26, 2016 at 4:43 AM, Tony Roberts <notifications@github.com>
wrote:

> In src/runtime/methodbinder.cs
> <https://github.com/pythonnet/pythonnet/pull/217#discussion_r64718246>:
>
> > @@ -344,7 +344,7 @@ internal Binding Bind(IntPtr inst, IntPtr args, IntPtr kw)
> >                              if (clrtype != null)
> >                              {
> >                                  bool typematch = false;
> > -                                if (pi[n].ParameterType != clrtype)
> > +                                if ((pi[n].ParameterType != clrtype) && !(pi[n].ParameterType.IsAssignableFrom(clrtype)))
>
> Isn't the != check redundant? It looks like checking
> !pi[n].ParameterType.IsAssignableFrom(clrtype) would be sufficient.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/pythonnet/pythonnet/pull/217/files/5a8ae4f6b1a69ad96c17d894972556296b7fe513#r64718246>
>